crates/biome_js_analyze/src/lint/nursery/no_restricted_types.rs

```diff
@@ -76,7 +76,15 @@ impl Rule for NoRestrictedTypes {
 
         let restricted_type = options.types.get(token_name)?.clone();
 
-        Some(restricted_type)
+        let resolved_restriction = match restricted_type {
+            CustomRestrictedType::Plain(message) => CustomRestrictedTypeOptions {
+                message,
+                use_instead: None,
+            },
+            CustomRestrictedType::WithOptions(options) => options,
+        };
```

Comment:
This could be placed in a `From<CustomRestrictedTypeOptions> for CustomRestrictedTypeOptions`.